pasture-core/src/gpu/device.rs

```diff
@@ -0,0 +1,639 @@
+use wgpu::util::{DeviceExt, BufferInitDescriptor};
+use wgpu::BufferDescriptor;
+use crate::layout::{PointAttributeDataType};
+use crate::layout;
+use crate::containers::{PointBuffer};
+use bytemuck::__core::convert::TryInto;
+
+pub struct Device {
+    // Private fields
+    adapter: wgpu::Adapter,
+    device: wgpu::Device,
+    queue: wgpu::Queue,
+
+    upload_buffers: Vec<wgpu::Buffer>,
```

Comment:
I like that with your approach, everything is neatly contained in the `Device` type. At the same time, I would have liked to have a separate `PointBuffer` type indicating a point buffer that is linked to the GPU, i.e. a buffer that has a corresponding memory region allocated on the GPU on which we can run compute shaders. 

I get that there are alignment problems on the GPU, especially that we can't use `Vec3<f32>` natively sucks, but maybe there is some other way to associate a CPU-side point buffer with a GPU-side point buffer. The whole upload/download process just doesn't seem super usable to me at the moment. Shouldn't `upload` at least return a handle to the point buffer that was uploaded, in case we want to upload multiple buffers? 

I don't have the answers, but I would like to discuss this further. Maybe we have to look at use-cases some more as well.